Makefile

```diff
@@ -0,0 +1,7 @@
+.PHONY: test
+test:
+	nosetests
```

Comment:
For now we're still using (cd utils; make ...); let's move utils/Makefile -> Makefile in a separate PR and we can have a more focused discussion of what runs when.